src/main/java/bot/telegram/state/StateProcessor.java

```diff
@@ -0,0 +1,109 @@
+package bot.telegram.state;
+
+import bot.domain.User;
+import bot.repository.UserService;
+import bot.telegram.keyboard.MainMenuKeyboard;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.telegram.telegrambots.meta.api.methods.send.SendMessage;
+import org.telegram.telegrambots.meta.api.objects.Message;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+public class StateProcessor {
+    MessageSender sender;
+    MainMenuKeyboard menu;
+    UserService service;
+
+    public SendMessage processMessage(BotState state, Message message) {
+        String chatId = message.getChatId().toString();
+        switch (state) {
+
+            case WANNA_GET_SHARE:
```

Comment:
@temalead Diverge from `switch` here also. It can perfectly be replaced using the solution I described above.

Reply:
You`re right, @Mikhail2048 I will refactor this code